src/lib/link.ts

```diff
@@ -345,6 +350,31 @@ export class Link {
     );
   }
 
+  public async getPendingAcks(
+    source: Side,
+    opts: QueryOpts = {}
+  ): Promise<AckWithMetadata[]> {
+    const { src, dest } = this.getEnds(source);
+    const allAcks = await src.queryWrittenAcks(opts);
+    if (allAcks.length === 0) {
+      return [];
+    }
+    const { sourcePort, sourceChannel } = allAcks[0].originalPacket;
+    // TODO: handle this when there are multiple channels,
+    const toCheck = allAcks.map((ack) =>
+      ack.originalPacket.sequence.toNumber()
+    );
+    const { sequences } = await dest.client.query.ibc.channel.unreceivedAcks(
+      sourcePort,
+      sourceChannel,
+      toCheck
+    );
+    const unreceived = new Map(sequences.map((seq) => [seq.toNumber(), true]));
```

Comment:
Why not use `new Set(sequences)` and test for membership with `unreceived.has(ack.originalPacket.sequence.toNumber())`?

Reply:
Can do. I did not know about that one

Reply:
Updated